event/consumer.go

```diff
@@ -47,31 +47,40 @@ func Consume(ctx context.Context, messageConsumer kafka.IConsumerGroup, handler
 // processMessage unmarshals the provided kafka message into an event and calls the handler.
 // After the message is handled, it is committed.
 func processMessage(ctx context.Context, message kafka.Message, handler Handler, cfg *config.Config) {
+	traceID := ctx.Value(kafka.TraceIDHeaderKey)
 	// unmarshal - commit on failure (consuming the message again would result in the same error)
 	event, err := unmarshal(message)
 	if err != nil {
-		log.Error(ctx, "failed to unmarshal event", err)
+		log.Error(ctx, "failed to unmarshal event", err, log.Data{
+			"request-id": traceID,
```

Comment:
COMMENT 1

You don't want to be adding the `request-id` in the log.Data. When the logs are passed to Kibana, it retrieves the traceID from the ctx (https://github.com/ONSdigital/dp-net/blob/main/request/context_key.go#L9) and adds it in the log visible in Kibana.

So by doing this, the trace id will be displayed twice in the Kibana logs

Reply:
Discussed and this implementation is fine